src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1721,14 +1720,27 @@ bool canProceed()
     return proceed;
 }
 
+static int isSudo()
```

Comment:
I like that you added this static method.  Could you refactor the code segment in the method xclP2p() 
```
    if (!root) {
        std::cout << "ERROR: root privileges required." << std::endl;
        return -EPERM;
    }
```
To use this method.

Reply:
you happened to find xclp2p(). When i did a search, I found some other places in the same file having same style.
I can fix this also. but I am not expected to fix this kind of issues, if any in the files, with this PR

Reply:
And this PR is just an ugly workaround to provide card level support on u30. There are many limitations to use this cmd, say root privilege required, not in container and/or VM, etc. 
This temp solution is for 2020.2. I don't think it should be in xbutil2